### DISCUSSION TITLE
cpp/memory/difference_of_pointers_and_ref

### DISCUSSION THREAD

**Discussion by giscus** [imarvinle/CSGuide#19] [Announcements]
# cpp/memory/difference_of_pointers_and_ref

编程指北: 计算机系统学习指南, 操作系统, 计算机网络, C++, Java, 算法, 数据结构

https://csguide.cn/cpp/memory/difference_of_pointers_and_ref.html

**Comment by kkk-455123** [comment]
北哥，最后总结中的第一条说引用可以看做是常量指针，引用指向不可变也就是指针本身不可变，是不是应该是指针常量？

**Reply by imarvinle** [reply]
对的，这里写得有误，我晚上回去修改下，或者你可以直接提一个MR，每个页面左下角点击就可以编辑了：![](https://cdn.how2cs.cn/csguide/121842.png)

**Comment by Taramesu** [comment]
速成速成，绝大多数都仅仅是速，你这个是真成了

**Comment by AaronPMM** [comment]
如何从汇编的角度上分析， p 和 r 打印的地址不同呢？因为汇编的代码一样的

**Comment by Ryan-216** [comment]
我哭死，居然真的有人和我一样去扒指针和引用的汇编代码，和同学说指针和引用在汇编层面完全一样他们还不信（TvT）

**Reply by imarvinle** [reply]
哈哈哈，因为当时学的时候就有这个疑问，所以看了下汇编，发现的确是编译器语法糖了

**Comment by Iwannnn** [comment]
编译器开启优化后引用有可能直接访问原变量
